### PR TITLE
Initial functional test script development

### PR DESCRIPTION
**Pull request by jlongever**
@RackHD/corecommitters @zyoung51 @tannoa2 @uppalk1 @keedya

- Initial CI test framework
- Module and test cleanup
- Add node API tests
- Added lookups and more amqp API tests
- Adding virtualenv and tweaking config for Vagrant env
- Rename directory
- Add env variable for host/port values
- Add module class comments
- Optional urls() headers argument
- Add loglevel env variable

### CODE REVIEW

File: test/tests/nodes_tests.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    'service': 'snmp-obm-service'
+                }]
+            },
+            {
+                'autoDiscover': 'false',
+                'name': 'test_pdu_node',
+                'type': 'pdu',
+                'obmSettings': [{
+                    'config': { 'host': '1.1.1.2', 'community': 'rackhd' },
+                    'service': 'snmp-obm-service'
```
Comment (benbp) [inline test/tests/nodes_tests.py]: 'snmp-obm-service' ?

Comment (jlongever) [inline test/tests/nodes_tests.py]: getting ahead of myself... how come the validation doesn't fail?

Comment (jlongever) [inline test/tests/nodes_tests.py]: nvm, cause autoDiscover is false.

**Comment by benbp** [comment]
:+1:

**Comment by tannoa2** [comment]
+1

**Pull request merged**